IDEA 2022.1 java.lang.NoClassDefFoundError

**Describe the bug**
IDEA 2022.1 java.lang.NoClassDefFoundError

**Additional context**

```
java.lang.NoClassDefFoundError
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:603)
	at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:678)
	at java.base/java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:722)
	at com.intellij.concurrency.JobLauncherImpl.processQueue(JobLauncherImpl.java:357)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.processInOrder(InspectionRunner.java:309)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.visitElements(InspectionRunner.java:284)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$inspect$5(InspectionRunner.java:122)
	at com.intellij.codeInspection.InspectionEngine.withSession(InspectionEngine.java:227)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.inspect(InspectionRunner.java:110)
	at com.intellij.codeInsight.daemon.impl.LocalInspectionsPass.collectInformationWithProgress(LocalInspectionsPass.java:132)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:97)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:56)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:419)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1152)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:412)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:411)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:387)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:174)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:213)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:385)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:184)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.xenoamess.p3c.pmd.lang.java.util.namelist.NameListConfig
	at com.alibaba.p3c.idea.compatible.inspection.Inspections.loadPatchConfigFile(Inspections.kt:56)
	at com.alibaba.p3c.idea.pmd.AliPmdProcessor.processFile(AliPmdProcessor.kt:56)
	at com.alibaba.p3c.idea.inspection.AliPmdInspectionInvoker.doInvoke(AliPmdInspectionInvoker.kt:62)
	at com.alibaba.p3c.idea.inspection.AliPmdInspectionInvoker$Companion.invokeInspection(AliPmdInspectionInvoker.kt:136)
	at com.alibaba.p3c.idea.inspection.AliPmdInspection.checkFile(AliPmdInspection.kt:71)
	at AvoidUseTimerRuleInspection.checkFile(DelegatePmdInspection.kt:50)
	at com.intellij.codeInspection.LocalInspectionTool$1.visitFile(LocalInspectionTool.java:141)
	at com.intellij.psi.impl.source.PsiJavaFileBaseImpl.accept(PsiJavaFileBaseImpl.java:483)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$processInOrder$10(InspectionRunner.java:333)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1152)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$processInOrder$11(InspectionRunner.java:312)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:174)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:213)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$processInOrder$12(InspectionRunner.java:346)
	at com.intellij.util.AstLoadingFilter.forceAllowTreeLoading(AstLoadingFilter.java:159)
	at com.intellij.util.AstLoadingFilter.forceAllowTreeLoading(AstLoadingFilter.java:151)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$processInOrder$13(InspectionRunner.java:310)
	at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:130)
	at com.intellij.util.AstLoadingFilter.disallowTreeLoading(AstLoadingFilter.java:119)
	at com.intellij.codeInsight.daemon.impl.InspectionRunner.lambda$processInOrder$14(InspectionRunner.java:310)
	at com.intellij.concurrency.JobLauncherImpl$1MyProcessQueueTask.lambda$call$0(JobLauncherImpl.java:297)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.concurrency.JobLauncherImpl$1MyProcessQueueTask.call(JobLauncherImpl.java:282)
	at com.intellij.concurrency.JobLauncherImpl$1MyProcessQueueTask.call(JobLauncherImpl.java:270)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedCallable.exec(ForkJoinTask.java:1448)
	... 5 more
```

--- a/idea-plugin/README.md
+++ b/idea-plugin/README.md
@@ -21,7 +21,7 @@ cd p3c-idea
 
 ## <font color="green">Use p3c-common as your plugin dependency</font>
 ```groovy
-compile 'com.xenoamess.p3c.idea:p3c-common:2.1.1.4x-SNAPSHOT'
+compile 'com.xenoamess.p3c.idea:p3c-common:2.1.1.4x'
 ```
 ## [中文使用手册](README_cn.md)
 ## <font color="green">Install</font>

--- a/idea-plugin/README.md
+++ b/idea-plugin/README.md
@@ -21,7 +21,7 @@ cd p3c-idea
 
 ## <font color="green">Use p3c-common as your plugin dependency</font>
 ```groovy
-compile 'com.xenoamess.p3c.idea:p3c-common:2.1.1.3x'
+compile 'com.xenoamess.p3c.idea:p3c-common:2.1.1.4x-SNAPSHOT'
 ```
 ## [中文使用手册](README_cn.md)
 ## <font color="green">Install</font>

--- a/idea-plugin/gradle.properties
+++ b/idea-plugin/gradle.properties
@@ -8,4 +8,4 @@ idea_version=2022.1
 #idea_version=145.258.11
 plugin_name=Alibaba Java Coding Guidelines
 systemProp.file.encoding=UTF-8
-plugin_version=2.1.1.4x-SNAPSHOT
+plugin_version=2.1.1.4x

--- a/idea-plugin/gradle.properties
+++ b/idea-plugin/gradle.properties
@@ -2,10 +2,10 @@
 # See https://www.jetbrains.org/intellij/sdk/docs/reference_guide/intellij_artifacts.html
 # See https://www.jetbrains.com/intellij-repository/releases/
 # See https://www.jetbrains.com/intellij-repository/snapshots/
-idea_version=201.7223.91
+idea_version=2022.1
 #idea_version=211-EAP-SNAPSHOT
 #idea_version=181.5540.23
 #idea_version=145.258.11
 plugin_name=Alibaba Java Coding Guidelines
 systemProp.file.encoding=UTF-8
-plugin_version=2.1.1.3x
+plugin_version=2.1.1.4x-SNAPSHOT

--- a/idea-plugin/p3c-common/build.gradle
+++ b/idea-plugin/p3c-common/build.gradle
@@ -60,7 +60,10 @@ ext.isReleaseVersion = !version.endsWith("SNAPSHOT")
 
 dependencies {
     implementation group: 'org.freemarker', name: 'freemarker', version: '2.3.31'
-    implementation "com.xenoamess.p3c:p3c-pmd:$plugin_version"
+    implementation("com.xenoamess.p3c:p3c-pmd:$plugin_version") {
+        exclude group: 'org.apache.logging.log4j'
+        exclude group: 'org.slf4j'
+    }
     implementation group: 'org.javassist', name: 'javassist', version: '3.28.0-GA'
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$project.kotlin_version"
     implementation "org.jetbrains.kotlin:kotlin-reflect:$project.kotlin_version"

--- a/idea-plugin/p3c-common/src/main/kotlin/com/alibaba/p3c/idea/inspection/standalone/AliMissingOverrideAnnotationInspection.kt
+++ b/idea-plugin/p3c-common/src/main/kotlin/com/alibaba/p3c/idea/inspection/standalone/AliMissingOverrideAnnotationInspection.kt
@@ -29,6 +29,7 @@ import com.intellij.psi.PsiMethod
 import com.intellij.psi.PsiModifier
 import com.intellij.psi.PsiModifierListOwner
 import com.intellij.psi.util.InheritanceUtil
+import com.jetbrains.rd.util.catch
 import com.siyeh.ig.BaseInspectionVisitor
 import com.siyeh.ig.InspectionGadgetsFix
 import com.siyeh.ig.inheritance.MissingOverrideAnnotationInspection
@@ -66,7 +67,12 @@ class AliMissingOverrideAnnotationInspection : MissingOverrideAnnotationInspecti
     override fun createOptionsPanel(): JComponent? = null
 
     override fun buildFix(vararg infos: Any): InspectionGadgetsFix? {
-        val fix = super.buildFix(*infos) ?: return null
+        val fix : InspectionGadgetsFix
+        try {
+            fix = super.buildFix(*infos) ?: return null
+        } catch (e: Exception){
+            return null;
+        }
         return DecorateInspectionGadgetsFix(
                 fix,
                 P3cBundle.getMessage("com.alibaba.p3c.idea.quickfix.standalone.AliMissingOverrideAnnotationInspection")

--- a/idea-plugin/p3c-idea/src/main/resources/META-INF/plugin.xml
+++ b/idea-plugin/p3c-idea/src/main/resources/META-INF/plugin.xml
@@ -29,7 +29,7 @@
     ]]></description>
 
     <change-notes><![CDATA[
-        <ul>2.1.1.4x-SNAPSHOT
+        <ul>2.1.1.4x
             <li>Fix idea 2022.1 Ultimate compatibility issue</li>
         </ul>
         <ul>2.1.1.3x

--- a/idea-plugin/p3c-idea/src/main/resources/META-INF/plugin.xml
+++ b/idea-plugin/p3c-idea/src/main/resources/META-INF/plugin.xml
@@ -29,6 +29,9 @@
     ]]></description>
 
     <change-notes><![CDATA[
+        <ul>2.1.1.4x-SNAPSHOT
+            <li>Fix idea 2022.1 Ultimate compatibility issue</li>
+        </ul>
         <ul>2.1.1.3x
             <li>pmd upgrade to 6.44.0</li>
             <li>fix bugs of SneakyThrowsWithoutExceptionTypeRule</li>

--- a/p3c-pmd/README.md
+++ b/p3c-pmd/README.md
@@ -11,7 +11,7 @@
 <dependency>
     <groupId>com.xenoamess.p3c</groupId>
     <artifactId>p3c-pmd</artifactId>
-    <version>2.1.1.4x-SNAPSHOT</version>
+    <version>2.1.1.4x</version>
 </dependency>
 ```
 ### <font color="green">Gradle</font>

--- a/p3c-pmd/README.md
+++ b/p3c-pmd/README.md
@@ -11,7 +11,7 @@
 <dependency>
     <groupId>com.xenoamess.p3c</groupId>
     <artifactId>p3c-pmd</artifactId>
-    <version>2.1.1.3x</version>
+    <version>2.1.1.4x-SNAPSHOT</version>
 </dependency>
 ```
 ### <font color="green">Gradle</font>

--- a/p3c-pmd/pom.xml
+++ b/p3c-pmd/pom.xml
@@ -6,7 +6,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.xenoamess.p3c</groupId>
     <artifactId>p3c-pmd</artifactId>
-    <version>2.1.1.3x</version>
+    <version>2.1.1.4x-SNAPSHOT</version>
     <packaging>jar</packaging>
     <name>p3c-pmd</name>
     <properties>

--- a/p3c-pmd/pom.xml
+++ b/p3c-pmd/pom.xml
@@ -6,7 +6,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.xenoamess.p3c</groupId>
     <artifactId>p3c-pmd</artifactId>
-    <version>2.1.1.4x-SNAPSHOT</version>
+    <version>2.1.1.4x</version>
     <packaging>jar</packaging>
     <name>p3c-pmd</name>
     <properties>

